crates/vault-registry/src/types.rs

```diff
@@ -369,12 +367,55 @@ impl<T: Config> RichVault<T> {
         self.decrease_issued(tokens)
     }
 
-    fn update_stake(&self) -> DispatchResult {
+    pub(crate) fn update_collateral_and_threshold(
```

Comment:
Rough implementation of [this](https://github.com/interlay/interbtc/pull/739/files#diff-faa164ec53eeb046689eebf0f2e1b0e48d7147c107fe8b0349c407d842d0713eR129-R152) function, needs to be called whenever the collateral or threshold changes and should update the stake in all three reward pools.